### PR TITLE
Replication e2e test

### PR DESCRIPTION
**Pull request by medegw01**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/270|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

```
# kubectl get pod -n dell-csm-operator
NAME                                                   READY   STATUS    RESTARTS   AGE
dell-csm-operator-controller-manager-97689b444-gmj74   2/2     Running   0          89s
```

Source
```
# kubectl get pods -n test-isilon
NAME                                     READY   STATUS    RESTARTS   AGE
test-isilon-controller-7b6667599-mwr42   6/6     Running   0          14s
test-isilon-controller-7b6667599-sxpb9   6/6     Running   0          14s
test-isilon-node-7p7sv                   2/2     Running   0          14s
test-isilon-node-mpnd6                   2/2     Running   0          14s
```

Target
```
# kubectl get pods -n test-isilon
NAME                                     READY   STATUS    RESTARTS   AGE
test-isilon-controller-79c59756f-hcgk9   6/6     Running   0          34s
test-isilon-controller-79c59756f-nsmp2   6/6     Running   0          34s
test-isilon-node-9d97k                   2/2     Running   0          34s
test-isilon-node-glbtp                   2/2     Running   0          34s
```

### CODE REVIEW

File: controllers/csm_controller.go [hunk 1]
```diff
@@ -717,6 +786,7 @@ func (r *ContainerStorageModuleReconciler) removeDriver(ctx context.Context, ins
 // PreChecks - validate input values
 func (r *ContainerStorageModuleReconciler) PreChecks(ctx context.Context, cr *csmv1.ContainerStorageModule, operatorConfig utils.OperatorConfig) error {
 	// Check drivers
+	fmt.Println(cr.Spec.Driver.CSIDriverType)
```
Comment (rensyct) [inline controllers/csm_controller.go:789]: Should this be removed?

File: pkg/modules/replication.go [hunk 2]
```diff
@@ -260,13 +257,13 @@ func ReplicationInjectClusterRole(clusterRole rbacv1.ClusterRole, cr csmv1.Conta
 func ReplicationPrecheck(ctx context.Context, op utils.OperatorConfig, replica csmv1.Module, cr csmv1.ContainerStorageModule, r utils.ReconcileCSM) error {
 	log := logger.GetLogger(ctx)
 
-	var repctlBinary string
+	//var repctlBinary string
```
Comment (rensyct) [inline pkg/modules/replication.go:260]: do we need this commented line

File: pkg/modules/replication.go [hunk 3]
```diff
@@ -260,13 +257,13 @@ func ReplicationInjectClusterRole(clusterRole rbacv1.ClusterRole, cr csmv1.Conta
...
 	if _, ok := ReplicationSupportedDrivers[string(cr.Spec.Driver.CSIDriverType)]; !ok {
 		return fmt.Errorf("CSM Operator does not suport Replication deployment for %s driver", cr.Spec.Driver.CSIDriverType)
 	}
 
 	// Check repctl binary runs fine
-	repctlBinary, ok := os.LookupEnv("REPCTL_BINARY")
+	/*repctlBinary, ok := os.LookupEnv("REPCTL_BINARY")
```
Comment (rensyct) [inline pkg/modules/replication.go:266]: Please remove the commented block

File: samples/storage_csm_powerscale_v220.yaml [hunk 4]
```diff
@@ -200,7 +200,7 @@ spec:
       #   true: enable replication feature(install dell-csi-replicator sidecar)
       #   false: disable replication feature(do not install dell-csi-replicator sidecar)
       # Default value: false
-      enabled: false
+      enabled: true
```
Comment (rensyct) [inline samples/storage_csm_powerscale_v220.yaml:203]: please set this to the default value which is false

**Review by rensyct — APPROVED** [review]

File: tests/e2e/steps/steps_def.go [hunk 5]
```diff
@@ -228,105 +163,154 @@ func (step *Step) validateModuleNotInstalled(res Resource, module string) error
...
-	err = step.ctrlClient.Get(context.TODO(), client.ObjectKey{
-		Namespace: cr.Namespace,
-		Name:      fmt.Sprintf("%s-node", cr.Name)}, ds)
+	return nil
+}
+
+func (step *Step) validateReplicationNotInstalled(res Resource) error {
+	cr := res.CustomResource
+
+	/* TODO(Michael): explore better way to handle this instead of using hacks*/
```
Comment (tdawe) [inline tests/e2e/steps/steps_def.go:226]: Will this be addressed in this PR or in a future PR?

File: controllers/csm_controller.go [hunk 6]
```diff
@@ -615,96 +673,107 @@ func (r *ContainerStorageModuleReconciler) getDriverConfig(ctx context.Context,
...
-		if err = utils.DeleteObject(ctx, &driverConfig.Controller.Rbac.ClusterRoleBinding, cluster.ClusterCTRLClient); err != nil {
-			log.Errorw("error delete controller cluster role binding", "Error", err.Error())
-			return err
-		}
+	if err = utils.DeleteObject(ctx, driverConfig.ConfigMap, cluster.ClusterCTRLClient); err != nil {
+		log.Errorw("error delete configmap", "Error", err.Error())
+		return err
+	}
+
+	if err = utils.DeleteObject(ctx, driverConfig.Driver, cluster.ClusterCTRLClient); err != nil {
```
Comment (tdawe) [inline controllers/csm_controller.go:717]: Is this guaranteed that we delete the target driver only if the Operator initially created it? Could this cause any issue if we want to keep the driver installed on the target cluster?

**Pull request merged**